docs/testclient.md

```diff
@@ -31,6 +31,21 @@ application. Occasionally you might want to test the content of 500 error
 responses, rather than allowing client to raise the server exception. In this
 case you should use `client = TestClient(app, raise_server_exceptions=False)`.
 
+### Selecting the Async backend
+
+`TestClient.async_backend` is a dictionary which allows you to set the options
+for the backend used to run tests.  These options are passed to
+`anyio.start_blocking_portal()`.  By default, `asyncio` is used.
+
+To run `Trio`, set `async_backend["backend"] = "trio"`, for example:
+
+```python
+def test_app()
+    client = TestClient(app)
+    client.async_backend["backend"] = "trio"
```

Comment:
This is interesting. I like the way you can set the backend, tho the dictionary style isn't immediately obvious to me. Are there *other* keys can be set here?

Reply:
trio lets you configure you stuff like `clock=None, instruments=(), restrict_keyboard_interrupt_to_checkpoints=False`

asyncio lets you configure: use_uvloop, debug, policy https://anyio.readthedocs.io/en/stable/basics.html#backend-specific-options

Reply:
it could be defined

```python
@dataclass
class TrioOptions:
    backend: Literal["trio"] = "trio"
    clock: trio.abc.Clock | None = None
    instruments: Sequence[trio.abc.Instrument] = ()
    restrict_keyboard_interrupt_to_checkpoints: bool = False

@dataclass
class AsyncioOptions:
   backend: Literal["asyncio"] = "asyncio"
   debug: bool = False
   use_uvloop: bool = True
   policy: asyncio.AbstractEventLoopPolicy | None = None
```

Reply:
Maybe the `backend_options` argument to `anyio.run()` should be typed in this way and we just expose that in Starlette?

Reply:
I'm not sure it's best to define specific backend options to starlette.  anyio could add new backends (twisted?) or the existing backends could support new options with new anyio versions.  It seems valuable to me to be as agnostic to the version of anyio as possible.

Seems to me that the best approach would be to use a `TypedDict` in `anyio.run()` for `backend_options`.  Any thoughts @agronholm ?

Reply:
AnyIO won't even try to keep up with the options provided by upstream. Is there a problem with that?

Reply:
With that in mind, it seems simplest to me to keep the options as a dictionary with options referenced to https://anyio.readthedocs.io/en/stable/basics.html#backend-specific-options

What I have could probably be refined and documented better, though.